doc/source/index.rst

```diff
@@ -368,6 +389,10 @@ Example
       - avhrr-3
 
     min_coverage: 25
+    s3_config:
+      target: s3://name-of-the-bucket/
+      delete_files: true
```

Comment:
I would like to avoid having a configuration for s3 here, because if we decide to add other protocols, it will require still other config sections.
My wish is to keep things simple in the config, and in the code.
So as an alternative, could we just use the `output_dir` parameter for specifying the target protocol? Like
`output_dir: s3://name-of-the-bucket/some/dir/`?
Then, we can use the `staging_zone` parameter to set where the files are saved locally before the transfer.

Reply:
Didn't know about the staging zone, can't see it in the documentation. I'll have a look.

Reply:
The staging zone is used to rename files after the Satpy writer is done with them, and needs the file to remain in the same filesystem. So it can't be used in this case.

Reply:
The functionality of `staging_zone` and file renaming have been modified so that the S3 uploader can use `staging_zone` as temporary local storage.